output/cloud/expv2/output.go

```diff
@@ -233,3 +307,24 @@ func (o *Output) handleFlushError(err error) {
 		}
 	}
 }
+
+func (o *Output) tracingEnabled() bool {
+	// TODO(lukasz): Check if the `tracing` module is imported.
+	//
+	// We want to check if the `tracing` module was imported. If
+	// it wasn't, we don't need to collect any data related to tracing
+	// as there won't be any.
+	//
+	// We currently don't have a way to check if the tracing module
+	// was imported by the user.
```

Comment:
My initial thought is that `(2)` is not actually a requirement at all.

FIrst and foremost just because someone imports this doesn't mean that they are not or they are using it to do tracing. Especially on the `not` side.

In this version of k6 #3037 will be released which means that practically the whole `k6/experimental/tracing` can now be written entirely in js code. 

It also means that users can now create and tag metrics with metadata that looks like the one from k6/experimental/tracing. And they might want to do it for other protocols/APIS (including not HTTP ones, which I guess we are not going to support initially).

Given that and the fact that `K6_CLOUD_TRACES_ENABLED` flag I feel like this is needless optimization to try to not do some more processing that will just bites us later on.

Reply:
I am not really certain going through all the metrics and checking whether they have metadata will be all that much "compute" compared to everything else the output(s) need to do.

And we likely can do it in a single place, so that will be even less of an issue. 

By a single place I mean that output v2 will aggregate and on the side check that a given metric sample has metadata - if it does - send it to the insights output.

Even without that I expect this to be an insignificant amount of work.

Also, again - we will only to this for the users that have this enabled, and I expect they will likely want to use that.

Reply:
@Blinkuu maybe it's my misunderstanding, but my interpretation of `(2)` was that it would drop the requirement for the flag so if a user does `k6 cloud script.js` and it is importing the `tracing` module then we should enable this proxy of the metadata. If this is not required, then I agree with @mstoykov if the user has to explicit the flag there is no reason for doing more.

@mstoykov I'm not sure I'm following, do you mean we should enable the check of the metadata for all scripts that the cloud run? If not, then how do we decide when to enable it or not for the cloud runs without `(3)`?

Reply:
> Ideally, we would like to enable the request metadata output logic if all 3 conditions are met:

Given this sentence my only opinion is taht (2) is not really necessary for us to not have this enabled most of the times. 

*BUT*:
1. It probably shouldn't be limitted based on 1 module given that users can generate their own `trace_id` metadata as well after this release
2. Even if it is enabled for a test that does not have metadata - I doubt there is any performance problem to be going through all the samples and checking if there is metadata on them.

Reply:
The idea behind `K6_CLOUD_TRACES_ENABLED` is to hide `k6 x Tempo` behind a feature flag until it's stable enough we can make it `true` by default. Hence I brought up (2). If you think this is unnecessary, let's skip this and revisit it later if we see big enough utilization spikes. I would like to keep the `TODO` for now tho.